### PR TITLE
[SPARK-19181][Core] Fixing flaky "SparkListenerSuite.local metrics" 

### PR DESCRIPTION
**Pull request by attilapiros**
## What changes were proposed in this pull request?

Sometimes "SparkListenerSuite.local metrics" test fails because the average of executorDeserializeTime is too short. As @squito suggested to avoid these situations in one of the task a reference introduced to an object implementing a custom Externalizable.readExternal which sleeps 1ms before returning.   

## How was this patch tested?

With unit tests (and checking the effect of this change to the average with a much larger sleep time).

### CODE REVIEW

**Comment by squito** [comment]
Jenkins, add to whitelist

File: core/src/test/scala/org/apache/spark/scheduler/SparkListenerSuite.scala [hunk 1]
```diff
@@ -583,3 +587,23 @@ private class FirehoseListenerThatAcceptsSparkConf(conf: SparkConf) extends Spar
     case _ =>
   }
 }
+
+private class SlowDeserializable(var readDuration: Int)
+  extends Externalizable {
+
+  def this() = this(0)
```
Comment (vanzin) [inline core/src/test/scala/org/apache/spark/scheduler/SparkListenerSuite.scala]: Is this needed for deserialization? I thought it was ok to not have a default constructor.

Comment (vanzin) [reply core/src/test/scala/org/apache/spark/scheduler/SparkListenerSuite.scala]: Nevermind, thinking of `Serializable`.

"When an Externalizable object is reconstructed, an instance is created using the public no-arg constructor"

File: core/src/test/scala/org/apache/spark/scheduler/SparkListenerSuite.scala [hunk 2]
```diff
@@ -583,3 +587,23 @@ private class FirehoseListenerThatAcceptsSparkConf(conf: SparkConf) extends Spar
     case _ =>
   }
 }
+
+private class SlowDeserializable(var readDuration: Int)
```
Comment (vanzin) [inline core/src/test/scala/org/apache/spark/scheduler/SparkListenerSuite.scala]: Probably easier to make the sleep time hardcoded? Then a lot of this code goes away.

(And if doing that, the `@throws` annotations are most probably unnecessary.)

**Comment by vanzin** [comment]
LGTM. Merging to master / 2.3.

**Pull request closed without merging**